File: gateway/src/main/java/kuku/advbkm/gateway/models/TemplateModel.java
package kuku.advbkm.gateway.models;

import lombok.*;

import java.util.HashMap;
import java.util.List;


@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
@ToString
public class TemplateModel {
    private String id;
    private String name;
    private String creatorID;
    private List<String> bookmarks;
    private HashMap<String, TemplateField> struct; //name of field  : fieldStruct
}

File: ui/lib/models/TemplateModel.dart
import 'TemplateField.dart';

class TemplateModel {
  String id;
  String name;
  String creatorID;
  List<String> bookmarks;
  Map<String, TemplateField> struct; //name of field  : fieldStruct
  TemplateModel(
      this.id, this.name, this.creatorID, this.bookmarks, this.struct);
}
